### PR TITLE
PreExecutionBenchmark uses setup/teardown for testing infrastructure

### PR DESCRIPTION
**Pull request by andreidan**

### CODE REVIEW

File: sql/src/test/java/io/crate/testing/SQLExecutor.java [hunk 1]
```diff
@@ -128,6 +133,7 @@ public SQLExecutor build() {
             if (!blobTables.isEmpty()) {
                 schemas.put(BlobSchemaInfo.NAME, new BlobSchemaInfo(clusterService, new TestingBlobTableInfoFactory(blobTables)));
             }
+            File tempDir = createTempDir();
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/testing/SQLExecutor.java]: This tempDir is never deleted this way.

Maybe make the SQLExecutor closeable or make the user responsible for providing an environment/tempfolder ?

File: sql/src/test/java/io/crate/testing/SQLExecutor.java [hunk 2]
```diff
@@ -165,6 +171,21 @@ public SQLExecutor build() {
             );
         }
 
+        private File createTempDir() {
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/testing/SQLExecutor.java]: Could be static

File: sql/src/jmh/java/io/crate/analyze/PreExecutionBenchmark.java [hunk 3]
```diff
@@ -27,29 +27,50 @@
...
+    private SQLExecutor e;
+    private Statement selectStatement;
+    private Analysis selectAnalysis;
+    private UUID jobId;
+
+    @Setup
+    public void setup() {
+        threadPool = new TestThreadPool("testing");
+        e = SQLExecutor.builder(
+            new ClusterService(Settings.builder().put("cluster.name", "ClusterServiceTests").build(),
```
Comment (seut) [inline sql/src/jmh/java/io/crate/analyze/PreExecutionBenchmark.java]: why aren't you extending from CrateDummyClusterServiceUnitTest and use its clusterService?

Comment (mfussenegger) [inline sql/src/jmh/java/io/crate/analyze/PreExecutionBenchmark.java]: I think that's fine, this is a benchmark not a unittest. I was actually wondering if we should just accept that the threadPool isn't closed if it doesn't influence the benchmarks

**Pull request merged**